### PR TITLE
#432 added test to confirm default methods work

### PR DESCRIPTION
**Pull request by krzyk**
#432
* added test that checks that the example provided in the issue actually works in the latest qulice

### CODE REVIEW

**Comment by davvd** [comment]
@krzyk Thanks, someone will review your pull request soon

**Comment by davvd** [comment]
@pinaf this pull request is for you, please review

File: qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+package foo;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Simple.
+ * @version $Id $
+ * @author John Smith (john@example.com)
```
Comment (pinaf) [inline qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java]: @krzyk why john smith?

File: qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * Hello.
+ */
+package foo;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Simple.
+ * @version $Id $
```
Comment (pinaf) [inline qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java]: @krzyk extra whitespace here

File: qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java [hunk 3]
```diff
@@ -147,6 +153,38 @@ public void acceptsInstanceMethodReferences() throws Exception {
...
+        final StringWriter writer = new StringWriter();
+        org.apache.log4j.Logger.getRootLogger().addAppender(
+            new WriterAppender(new SimpleLayout(), writer)
+        );
+        final Environment env = mock.withParam(
+            CheckstyleValidatorTest.LICENSE_PROP,
+            this.toURL(license)
+        )
+            .withFile(
+                "src/main/java/foo/DefaultMethods.java",
```
Comment (pinaf) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @krzyk the path with `foo` doesn't seem right to me

**Comment by pinaf** [comment]
@krzyk 3 comments

File: qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java (hunk 1 above)
Comment (krzyk) [inline qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java]: @pinaf why not? :) This is just a text file that will be checked by checkstyle, not a production code.

File: qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java (hunk 3 above)
Comment (krzyk) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @pinaf why? this is a test class, that will be created in package `foo`

File: qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java (hunk 1 above)
Comment (krzyk) [inline qulice-checkstyle/src/test/resources/com/qulice/checkstyle/DefaultMethods.java]: @pinaf all other examples use similar `@author` tags

File: qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java (hunk 3 above)
Comment (pinaf) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @krzyk Oh I though this was loading that other file you created with the `default` method.

**Comment by krzyk** [comment]
@pinaf correction and my responses :)

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by krzyk** [comment]
@davvd we are waiting for #460

**Comment by krzyk** [comment]
@yegor256 please try again

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
> @davvd we are waiting for #460 

@krzyk #460 is closed already

**Comment by davvd** [comment]
@pinaf thanks, paid, **17 mins** to your account, payment ID is `AP-6F5747155E254694J`, task took 97 hours and 11 mins to complete; review comments (c=2) added as a bonus; +17 added to your rating, at the moment it is: [+4069](http://www.netbout.com/b/35128?open=rating)

**Comment by davvd** [comment]
@rultor please deploy

**Pull request merged**